scripts/protect_branch_git_hook.py

```diff
@@ -0,0 +1,31 @@
+import re
+import subprocess
+import sys
+from typing import NoReturn
+
+
+def ProtectBranches() -> NoReturn:
+    hookid = "protect-branches"
+    protected_branches = [r"main", r"branch-\d+\.\d+"]
+    current_branch = (
+        subprocess.run(["git", "branch", "--show-current"], capture_output=True)
+        .stdout.decode(sys.stdout.encoding)
+        .replace("\n", "")
+    )
+    for branch in protected_branches:
+        regex = re.compile(branch)
+        if regex.match(current_branch):
+            # Not portable to get user input, see:
+            # https://stackoverflow.com/questions/65844278/any-way-to-get-user-input-from-a-hook
+            print(
+                f"You were about to push to `{current_branch}`, if that's what you intended, "
+                "you should run the following command:\n"
+                f"SKIP={hookid} git push"
+            )
```

Comment:
Triple quoted string would more align with the practice in the rest of the codebase:

```suggestion
            print(f"""
You were about to push to `{current_branch}`, which is disallowed by default.
If that's really what you intend, run the following command:

        SKIP={hookid} git push
""")
```

Though, speaking of portability, AFAIK the above will only work on OSX and Linux? cc @tcmetzger who might know how to issue the same command on windows